src/realm/table.cpp

```diff
@@ -2157,6 +2157,42 @@ void Table::insert_empty_row(size_t row_ndx, size_t num_rows)
     }
 }
 
+size_t Table::add_row_with_key(size_t key_col_ndx, int64_t key)
+{
+    size_t num_cols = m_spec.get_column_count();
+    size_t row_ndx = m_size;
+
+    REALM_ASSERT(is_attached());
+    REALM_ASSERT_3(key_col_ndx, <, num_cols);
```

Comment:
We could also assert on the type of the column we're inserting the integer into.